recursion.rb

```diff
@@ -1,19 +1,154 @@
-
+# Write #1
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  else
+    return n * fact(n-1)
+  end
 end
 
+# Write #2
 def reverse(s)
+  if s.length == 0 || s.length == 1
+    return s
+  else
+    return s[s.length - 1] + reverse(s.slice(0, s.length-1))
+  end
+end
+puts reverse("hello")
+# => "olleh"
+
+# Write #3
+def bunny(n)
+  if n == 0
+    return 0
+  end
+  return 2 + bunny(n-1)
 end
 
+# Write #4
 def nested(s)
+  if  s.length == 0
+    return true
+  end
+  if s.length % 2 != 0
+    return false
+  end
+  if s[0] != "("
+    return false
+  end
+  if s[s.length - 1] != ")"
+    return false;
+  end
+  return nested(s.slice(1, s.length - 2))
+end
+puts nested("((()))")
+# => true
+puts nested("())")
+# => false
+
+# fib(n)
+def fib(n)
+  if n == 0 || n == 1
+    return n
+  end
+  return fib( n - 1 ) + fib( n - 2 )
+end
+
+puts fib(4)
+# => 3
+
+# pal(s)
+def pal(s)
+  if  s.length == 0 || s.length == 1
+    return true
+  end
+  if s[0] != s[s.length - 1]
+    return false
+  end
+  return pal(s.slice(1, s.length - 2))
+end
```

Comment:
👍 

Could have used elsif's to make this slightly more readable, but it works and looks good!